src/datetime/FormatString.php

```diff
@@ -0,0 +1,72 @@
+<?hh
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace HH\Lib\Experimental\DateTime {
+  /**
+   * strftime(3)-compatible format string that includes standard placeholders
+   * like `%y` for year, but does not include timezone placeholders like `%z`.
+   */
+  type UnzonedDateFormatString = \HH\FormatString<_Private\DateFormat>;
+
+  /**
+   * strftime(3)-compatible format string that includes standard placeholders
+   * like `%y` for year, as well as timezone placeholders like `%z`.
+   */
+  type ZonedDateFormatString = \HH\FormatString<_Private\ZonedDateFormat>;
+}
+
+namespace HH\Lib\Experimental\DateTime\_Private {
```

Comment:
ah yep :)

Also, split out to separate file, at least one per NS if not per class.

- generally good to separate private from non-private where possible
- we currently have tools (e.g. the flib wrapper generator) that messes up if there are multiple NSes in a file